File: week_1/day-6-mixed-practice/follow-up/longest_subarray_positive_numbers.py
# Program: longest_subarray_positive_numbers.py

def longestSubarrayPositive(nums, k):
    left = 0
    curr_sum = 0
    max_len = 0

    for right in range(len(nums)):
        curr_sum += nums[right]

        while curr_sum > k:
            curr_sum -= nums[left]
            left += 1

        if curr_sum == k:
            max_len = max(max_len, right - left + 1)

    return max_len


# INPUT
nums = [1, 2, 1, 1, 1]
k = 3
print(longestSubarrayPositive(nums, k))  # Output: 3

File: week_1/day-6-mixed-practice/follow-up/longestSubarrayPositiveNumbers.js
// Program: longestSubarrayPositiveNumbers.js

function longestSubarrayPositive(nums, k) {
    let left = 0, sum = 0, maxLen = 0;

    for (let right = 0; right < nums.length; right++) {
        sum += nums[right];

        while (sum > k) {
            sum -= nums[left];
            left++;
        }

        if (sum === k) {
            maxLen = Math.max(maxLen, right - left + 1);
        }
    }
    return maxLen;
}

// INPUT
console.log(longestSubarrayPositive([1, 2, 1, 1, 1], 3)); // 3
